src/main/java/com/dataloom/organization/OrganizationsApi.java

```diff
@@ -1,5 +1,153 @@
 package com.dataloom.organization;
 
+import com.dataloom.authorization.Principal;
+import com.dataloom.authorization.PrincipalType;
+import retrofit2.http.*;
+
+import java.util.Set;
+import java.util.UUID;
+
 public interface OrganizationsApi {
+    String ORGANIZATIONS     = "organizations";
+    String ID                = "id";
+    String ID_PATH           = "/{" + ID + "}";
+    String DESCRIPTION       = "/description";
+    String TITLE             = "/title";
+    String EMAIL_DOMAIN      = "email-domain";
+    String EMAIL_DOMAINS     = "/email-domains";
+    String EMAIL_DOMAIN_PATH = "/{" + EMAIL_DOMAIN + "}";
+    String PRINCIPALS        = "/principals";
+    String PRINCIPAL_ID      = "pid";
+    String PRINCIPAL_ID_PATH = "/{" + PRINCIPAL_ID + "}";
+    String TYPE              = "type";
+    String TYPE_PATH         = "/{" + TYPE + "}";
+    String ROLES             = "/roles";
+    String MEMBERS           = "/members";
+
+    @GET( ORGANIZATIONS )
+    Iterable<Organization> getOrganizations();
+
+    @POST( ORGANIZATIONS )
+    UUID createOrganizationIfNotExists( @Body Organization organization );
+
+    @GET( ORGANIZATIONS + ID_PATH )
+    Organization getOrganization( @Path( ID ) UUID organizationId );
+
+    @DELETE( ORGANIZATIONS + ID_PATH )
+    Void destroyOrganization( @Path( ID ) UUID organizationId );
+
+    @PUT( ORGANIZATIONS + ID_PATH + TITLE )
+    Void updateTitle( @Path( ID ) UUID organziationId, @Body String title );
```

Comment:
Cleaner than creating an object for both. Humans are unlikely to edit both simultaneously and puts will be efficient ways of constantly saving.